internal/featuretests/v3/gatewayapi_test.go

```diff
@@ -0,0 +1,160 @@
+	})
+
+	rh.OnAdd(&gatewayapi_v1alpha1.Gateway{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      "contour",
+			Namespace: "projectcontour",
+		},
+		Spec: gatewayapi_v1alpha1.GatewaySpec{
+			Listeners: []gatewayapi_v1alpha1.Listener{{
+				Port:     80,
+				Protocol: "HTTP",
+				Routes: gatewayapi_v1alpha1.RouteBindingSelector{
+					Namespaces: gatewayapi_v1alpha1.RouteNamespaces{
+						From: gatewayapi_v1alpha1.RouteSelectAll,
+					},
+					Kind: dag.KindHTTPRoute,
+				},
+			}, {
+				Port:     443,
+				Protocol: "HTTPS",
+				TLS: &gatewayapi_v1alpha1.GatewayTLSConfig{
+					CertificateRef: &gatewayapi_v1alpha1.LocalObjectReference{
+						Group: "core",
+						Kind:  "Secret",
+						Name:  "tlscert",
+					},
+				},
+				Routes: gatewayapi_v1alpha1.RouteBindingSelector{
+					Namespaces: gatewayapi_v1alpha1.RouteNamespaces{
+						From: gatewayapi_v1alpha1.RouteSelectAll,
+					},
+					Kind: dag.KindHTTPRoute,
+				},
+			}},
+		},
+	})
+
+	rh.OnAdd(&gatewayapi_v1alpha1.HTTPRoute{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      "basic",
+			Namespace: "default",
+			Labels: map[string]string{
+				"app":  "contour",
+				"type": "controller",
+			},
+		},
+		Spec: gatewayapi_v1alpha1.HTTPRouteSpec{
+			Hostnames: []gatewayapi_v1alpha1.Hostname{
+				"test.projectcontour.io",
+			},
+			Rules: []gatewayapi_v1alpha1.HTTPRouteRule{{
+				Matches: []gatewayapi_v1alpha1.HTTPRouteMatch{{
+					Path: gatewayapi_v1alpha1.HTTPPathMatch{
+						Type:  "Prefix",
+						Value: "/blog",
+					},
+				}},
+				ForwardTo: []gatewayapi_v1alpha1.HTTPRouteForwardTo{{
+					ServiceName: pointer.StringPtr("svc2"),
+					Port:        gatewayPort(80),
+				}},
+			}, {
+				Matches: []gatewayapi_v1alpha1.HTTPRouteMatch{{
+					Path: gatewayapi_v1alpha1.HTTPPathMatch{
+						Type:  "Prefix",
+						Value: "/",
+					},
+				}},
+				ForwardTo: []gatewayapi_v1alpha1.HTTPRouteForwardTo{{
+					ServiceName: pointer.StringPtr("svc1"),
+					Port:        gatewayPort(80),
+				}},
+			}},
+		},
+	})
+
+	c.Request(routeType).Equals(&envoy_discovery_v3.DiscoveryResponse{
+		Resources: resources(t,
+			envoy_v3.RouteConfiguration("https/test.projectcontour.io",
+				envoy_v3.VirtualHost("test.projectcontour.io",
+					&envoy_route_v3.Route{
+						Match:  routePrefix("/blog"),
+						Action: routeCluster("default/svc2/80/da39a3ee5e"),
+					}, &envoy_route_v3.Route{
+						Match:  routePrefix("/"),
+						Action: routeCluster("default/svc1/80/da39a3ee5e"),
+					},
+				),
+			),
+			envoy_v3.RouteConfiguration("ingress_http",
+				envoy_v3.VirtualHost("test.projectcontour.io",
+					&envoy_route_v3.Route{
+						Match:  routePrefix("/blog"),
+						Action: routeCluster("default/svc2/80/da39a3ee5e"),
+					}, &envoy_route_v3.Route{
+						Match:  routePrefix("/"),
+						Action: routeCluster("default/svc1/80/da39a3ee5e"),
+					},
+				),
+			),
+		),
+		TypeUrl: routeType,
+	})
```

Comment:
worth checking the secrets returned include the configured secret?

Reply:
👍